src/main/kotlin/org/wycliffeassociates/otter/common/collections/FloatRingBuffer.kt

```diff
@@ -0,0 +1,76 @@
+package org.wycliffeassociates.otter.common.collections
+
+
+/**
+ * Created by sarabiaj on 2/21/2017.
+ */
+class FloatRingBuffer(capacity: Int) {
+
+    private var head = 0
+    private var tail = 0
+    private val buffer = FloatArray(capacity)
+    private val returnBuffer = FloatArray(capacity * 4)
+
+
+    val isEmpty: Boolean
+        get() = head == tail
+
+    val array: FloatArray
+        @Synchronized
+        get() {
+            var i = 0
+            while (i < size()) {
```

Comment:
This doesn't hurt much, but if you're really after performance, don't keep calculating size. It shouldn't change during this loop.